src/modal/index.tsx

```diff
@@ -158,6 +158,10 @@ class Modal extends Component<ModalProps> {
       </div>
     )
 
+    if (!canUseDOM || !this.portalNode) {
+      return modalElement
```

Comment:
I don't think this will work the modal element won't be at the same place as it is with the portal.
Did you check your change in Kairos ?

Reply:
I tested everything in Kairos, this works ;)

Reply:
From what I understand of the way `portal` works, is that even if the element is in another `<div>`, React still considers it at its initial position in the tree, which explains the rendering issue in the first place.

Reply:
Add some test by moking canUseDom to be sure you would render it properly server side. And avoid future regressions.

Reply:
I can't mock it, I tried several times, but it always stays to true...